src/iarc7_motion/iarc_tasks/go_to_roomba_task.py

```diff
@@ -0,0 +1,186 @@
+                            tf2_ros.ExtrapolationException) as ex:
+                        rospy.logerr('ObjectTrackTask: Exception when looking up transform')
+                        rospy.logerr(ex.message)
+                        return (TaskAborted(msg='Exception when looking up transform during go to roomba'),)
+                        
+
+
+                    # Create point at drones center
+                    stamped_point = PointStamped()
+                    stamped_point.point.x = 0
+                    stamped_point.point.y = 0
+                    stamped_point.point.z = 0
+
+                    # returns point distances of roomba to center point of the map
+                    self._roomba_point_map = tf2_geometry_msgs.do_transform_point(
+                                                            stamped_point, roomba_transform)
+
+                    self._roomba_point_level_quad = tf2_geometry_msgs.do_transform_point(
+                                                            stamped_point, roomba_check_pos)
+
+
+                    roomba_x_velocity = self._roomba_odometry.twist.twist.linear.x
+                    roomba_y_velocity = self._roomba_odometry.twist.twist.linear.y
+
+                    x_vel_target = self._roomba_point_map.point.x + roomba_x_velocity
+                    y_vel_target = self._roomba_point_map.point.y + roomba_y_velocity
+                    
+                    self._path_holder.reinitialize_translation_stop_planner(x_vel_target,
+                                                                            y_vel_target,
+                                                                            self._z_position)
+
+
+                    """ quarantine end"""
+                    hold_twist = self._path_holder.get_xyz_hold_response()
+                    if self._check_max_roomba_range():
+                        return (TaskDone(), VelocityCommand(hold_twist))
+                    elif not self._path_holder.is_done():
+                        return (TaskRunning(), VelocityCommand(hold_twist))
+                    else:
+                        return (TaskDone(), VelocityCommand(hold_twist))
+                else:
+                    return (TaskFailed(msg='Fell below minimum manuever height during translation'),)
+
+
+            return (TaskAborted(msg='Impossible state in takeoff task reached'))
+
+    """ qurantine methods """
+    def _check_roomba_in_sight(self):
+        for odometry in self.topic_buffer.get_roomba_message().data:
+            if odometry.child_frame_id == self._roomba_id:
+                self._roomba_odometry = odometry
+                return True
+        return False
+
+    def _check_max_roomba_range(self,):
+        _distance_to_roomba = math.sqrt(self._roomba_point_level_quad.point.x**2 +
+                            self._roomba_point_level_quad.point.y**2)
+        return (_distance_to_roomba <= self._MAX_TASK_DIST * 0.75)
+
+    def _on_ground(self):
```

Comment:
We should not be checking if we are on the ground in this task, as it shouldn't be landing the drone.

Reply:
Removed